webapp/src/jaas/components/conference/jaas_conference.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import {Client4} from 'mattermost-redux/client';
+import {id as pluginId} from '../../../manifest';
+import constants from '../../../constants';
+
+type Props = {}
+
+type State = {
+    jaasJwt?: string | null,
+    jaasRoom?: string | null,
+}
+
+const JWT = 'jwt';
+const JAAS_URL = '8x8.vc';
+
+export default class JaaSConference extends React.PureComponent<Props, State> {
+    api: any;
+    style = getStyle();
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {};
+    }
+
+    initJaaS(jwt: string, room: string) {
+        const ws = Math.max(document.documentElement.clientWidth ?? 0, window?.innerWidth ?? 0);
+        const hs = Math.max(document.documentElement.clientHeight ?? 0, window?.innerHeight ?? 0);
+        const options = {
+            width: ws,
+            height: hs,
+            roomName: room,
+            jwt,
+            parentNode: document.querySelector('#jitsiMeet')
+        };
+
+        this.api = new (window as any).JitsiMeetExternalAPI(JAAS_URL, options);
+    }
+
+    startJaaSMeetingWindow(jwt: string | null, meetingId: string | null) {
+        const options = {
+            method: 'post',
+            body: JSON.stringify({jaasJwt: jwt ?? '', jaasPath: meetingId}),
+            headers: {}
+        };
+
+        const baseUrl = `/plugins/${pluginId}`;
+        fetch(`${baseUrl}/api/v1/meetings/jaas/settings`, Client4.getOptions(options)).
+            then((result) => {
+                if (result.ok) {
+                    return result.json();
+                }
+
+                throw new Error(result.statusText);
+            }).
+            then((result) => {
+                this.setState({
+                    jaasRoom: result.jaasRoom,
+                    jaasJwt: result.jaasJwt
+                });
+            });
+    }
+
+    componentDidMount() {
+        if (!(window as any).JitsiMeetExternalAPI) {
+            const script = document.createElement('script');
+            script.type = 'text/javascript';
+            script.onload = () => {
```

Comment:
This component could mount more than once over the course of a browser session. There may be implications if we load this script more than once, so we should avoid check if it's already been loaded before loading again.

Reply:
@mickmister Added the condition to check whether the script is already loaded or not.